src/loader/ident_env.cppo.ml

```diff
@@ -571,18 +574,18 @@ module Path = struct
   let read_class_type env = function
     | Path.Pident id -> read_class_type_ident env id
 #if OCAML_VERSION >= (4,8,0)
-    | Path.Pdot(p, s) -> `Dot(read_module env p, s)
+    | Path.Pdot(p, s) -> `Dot(read_module env p, strip_hash s)
```

Comment:
We could have a function for that (`read_type_path p s`) and a bit of documentation. Also should it be called in `Fragment.read_type` ?

Reply:
Good point, I'll check for fragments. Do you mean a function to return `` `Dot(...) ``? Might have to have some annoying coercions, but I can see how it looks.

Reply:
> Do you mean a function to return `Dot(...)?

Yes but if the type is annoying, that's not a big deal. Perhaps abstracting `strip_hash` behind a `read_type_name s` would be enough to justify it.

Reply:
I don't think you can specify '#' in fragments, so that's probably fine.

Reply:
and I'm not keen on the function for reading `` `Dot ``, mostly because we wouldn't want to call it from `read_module` and `read_module_type`. I'll add some documentation though.